modules/phase_field/examples/traveling_wave/double_obstacle.i

```diff
@@ -0,0 +1,132 @@
+# example of traveling wave solution for double obstacle poetential
+# governing equation:
+#   tao * phi_dot = epsilon * laplace(phi) + gamma * (phi - 0.5) + sqrt(phi * (1 - phi)) * m;
+# coefficients:
+#   epsilon = 8 * sigma * eta / pi / pi, gamma = 8 * sigma / eta,
+#   tao = 8 * eta / mu / pi / pi,        m = -8 / pi * delta_g
+# parameters:
+#   delta_g = 1.0, eta = 6.0, mu = 1.0, sigma = 1.0
+# reference:
+#   I. Steinbach, Modelling and Simulation in Materials Science and Engineering. 17(7) (2009) 073001.
+#   website: https://iopscience.iop.org/article/10.1088/0965-0393/17/7/073001/pdf 
+# analytical solution:
+#   Eq.(67) in the reference
+
+[Mesh]
+  type = GeneratedMesh
+  dim = 1
+  nx = 100
+  xmin = -50
+  xmax = 50
+  elem_type = EDGE
+[]
+
+[Variables]
+  [./phi]
+    family = LAGRANGE
+    order = FIRST
+  [../]
+[]
+
+[ICs]
+  [./ICc]
+    type = FunctionIC
+    variable = phi
+    function = "if(x < 0, 0, 1)"
+  [../]
+[]
+
+[Kernels]
+  [./tws_diffusion]
+    type = NonlinearDiffusion
+    variable = phi
+    D = epsilon
+  [../]
+  [./tws_time_derivative]
+    type = NonlinearTimeDerivative
+    variable = phi
+    coefficient = tao
+  [../]
```

Comment:
```suggestion
  [tws_diffusion]
    type = MatDiffusion
    variable = phi
    diffusivity = epsilon
  []
  [tws_time_derivative]
    type = SusceptibilityTimeDerivative
    variable = phi
    f_name = tao
  []
```

should work without automatic differentiation (if you provide derivatives of `tao` using the derivative function material interface)